group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeAssignor.java

```diff
@@ -0,0 +1,228 @@
+     * <ol>
+     * <li> Generate a map of members per topic using the given member subscriptions.</li>
+     * <li> Generate a list of members called potentially unfilled members, which consists of members that have not met the minimum required quota of partitions for the assignment AND
+     * get a list called assigned sticky partitions per topic, which has the partitions that will be retained in the new assignment.</li>
+     * <li> Generate a list of unassigned partitions by calculating the difference between the total partitions for the topic and the assigned (sticky) partitions. </li>
+     * <li> Find members from the potentially unfilled members list that haven't met the total required quota
+     * i.e. minRequiredQuota + 1, if the member is designated to receive one of the excess partitions OR minRequiredQuota otherwise. </li>
+     * <li> Assign partitions to them in ranges from the unassigned partitions per topic
+     *      based on the remaining partitions value. </li>
+     * </ol>
+     * </p>
+     */
+    @Override
+    public GroupAssignment assign(final AssignmentSpec assignmentSpec) throws PartitionAssignorException {
+        Map<String, MemberAssignment> newAssignment = new HashMap<>();
+
+        // Step 1
+        Map<Uuid, List<String>> membersPerTopic = membersPerTopic(assignmentSpec);
+
+        membersPerTopic.forEach((topicId, membersForTopic) -> {
+
+            AssignmentTopicMetadata topicData = assignmentSpec.topics().get(topicId);
+            int numPartitionsForTopic = topicData.numPartitions();
+            int minRequiredQuota = numPartitionsForTopic / membersForTopic.size();
+            // Each member can get only ONE extra partition per topic after receiving the minimum quota.
+            int numMembersWithExtraPartition = numPartitionsForTopic % membersForTopic.size();
+
+            // Idle members case : When the number of members subscribed to a topic is greater than the total number of partitions,
+            // all members get assigned via the "extra partitions" logic since minRequiredQuota = 0.
+
+            // Step 2
+            Set<Integer> assignedStickyPartitionsForTopic = new HashSet<>();
+            List<MemberWithRemainingAssignments> potentiallyUnfilledMembers = new ArrayList<>();
+
+            for (String memberId : membersForTopic) {
+                Set<Integer> assignedPartitionsForTopic = assignmentSpec.members().get(memberId)
+                    .assignedPartitions().getOrDefault(topicId, Collections.emptySet());
+
+                int currentAssignmentSize = assignedPartitionsForTopic.size();
+                List<Integer> currentAssignmentListForTopic = new ArrayList<>(assignedPartitionsForTopic);
+
+                // If there were partitions from this topic that were previously assigned to this member present, retain as much as possible.
+                // We need to retain:
+                // 1) currentSize number of partitions (when currentSize < required) OR
```

Comment:
`currentSize` does not exist any more. This is why I don't like to use variable names in comments :)

Reply:
what are we supposed to use instead?

Reply:
okay I wont use variable names again